sdc11073/sdcdevice/httpserver.py

```diff
@@ -173,14 +173,17 @@ def do_POST(self):
                     response_xml_string = devices_dispatcher.on_post(self.path, self.headers, request)
                     http_status = 200
                     http_reason = 'Ok'
+                    # MDPWS:R0007 A text SOAP envelope shall be serialized using utf-8 character encoding
+                    assert (b'utf-8' in response_xml_string[:100].lower())
                 except HTTPRequestHandlingError as ex:
                     response_xml_string = ex.soapfault
                     http_status = ex.status
                     http_reason = ex.reason
 
                 commlog.defaultLogger.logSoapRespOut(response_xml_string, 'POST')
                 self.send_response(http_status, http_reason)
-            assert(b'utf-8' in response_xml_string[:100].lower()) # MDPWS:R0007 A text SOAP envelope shall be serialized using utf-8 character encoding
+            if isinstance(response_xml_string, str):
```

Comment:
Why not make the response_xml_string literal above into bytes, so you don't have to do this conversion if there is no dispatcher? Seems to me like it's the only case in which this is actually a string